### ISSUE TITLE
[Code Generation] Add support for directives

### ISSUE THREAD

**Issue by marwanad** [apollostack/apollo-android#28]
Add support for `@skip` and `@include` directives

```
query HeroNameInclusionDirective {
     hero {
          name @include(if: someCondition)
        }
}
```

There would be an `isConditional` field  in the IR and handling those should be somewhat similar to handling nullability.

**Comment by marwanad** [comment]
Taking this issue up next.

**Comment by felipecsl** [comment]
Fixed via #30
